File: src/animal.py
import random as rd

class Animal :

    def __init__(self, energy_, position_, velocity_, max_velocity_, color_) :
        self.energy = energy_
        self.position = position_
        self.velocity = velocity_
        self.max_velocity = max_velocity_

    def move(self, next_position = None) :
        if next_position is None :
            next_position = [coord + rd.randint() for coord in self.position]
        self.position = next_position

    def eat() :
        pass

    def reproduce() :
        pass

    def die() :
        pass

File: c++/src/Animal.cpp
#pragma once
#include <SFML/Graphics.hpp>

#define RADIUS 10

class Animal {
    public:
        int radius = RADIUS;
        int energy;
        sf::Vector2f position;
        sf::Vector2f velocity;
        bool is_dead;
        
        Animal(int energy_, sf::Vector2f position_, sf::Vector2f velocity_) : 
            energy(energy_),
            position(position_),
            velocity(velocity_),
            is_dead(false)
        {};

        void move() {
            position += velocity;
        }

        void eat() {
        }

        void reproduce() {
        }

        void die() {
            is_dead = (energy <= 0);
        }
        
};